netlogo-gui/src/main/swing/BrowserLauncher.scala

```diff
@@ -0,0 +1,124 @@
+// (C) Uri Wilensky. https://github.com/NetLogo/NetLogo
+
+package org.nlogo.swing
+
+import java.awt.{ Component, Desktop }
+import java.lang.Process
+import java.io.IOException
+import java.net.{ URI, URISyntaxException, URLEncoder }
+import java.nio.file.{ Files, Path, Paths }
+import javax.swing.JOptionPane
+
+object BrowserLauncher {
+  private val redirectUri = docPath("redirect.html").toUri
+  private val osName = System.getProperty("os.name")
+
+  @deprecated("Use openURI or openPath instead", "6.0.2")
+  def openURL(comp: Component, urlString: String, local: Boolean): Unit = {
+    if (local)
+      openPath(comp, Paths.get(urlString), "")
+    else
+      Option(makeURI(comp, urlString)).foreach(openURI(comp, _))
+  }
+
+  def openURI(comp: Component, uri: URI): Unit = {
+    try {
+      val opened =
+        desktop.map(d => {d.browse(uri); true}).getOrElse(false)
+      if (! opened)
+        osSpecificBrowserRunner.getOrElse(browserNotFound()).apply(uri.toString)
+    } catch {
+      case ex: BrowserNotFoundException =>
+        JOptionPane.showMessageDialog(comp, ex.getLocalizedMessage)
+      case ex: IOException =>
+        JOptionPane.showMessageDialog(comp, s"Unable to open a browser to: ${uri.toString}\n" +
+          "Please report to bugs@ccl.northwestern.edu")
+    }
+  }
+
+  def openPath(comp: Component, path: Path, anchor: String): Unit = {
+    val u = path.toUri
+    if (anchor == null || anchor == "")
+      openURI(comp, u)
+    else {
+      // Windows and Linux don't support anchors in file URLs, so we create a redirect helper
+      // to work around this.
+      val uriWithAnchor = new URI(u.getScheme, u.getHost, u.getPath, anchor)
+      val redirectFile = Files.createTempFile("redirectHelper", ".html")
+      Files.write(redirectFile, redirectHelper(uriWithAnchor).getBytes("UTF-8"))
+      openURI(comp, redirectFile.toUri)
+    }
+  }
+
+  def redirectHelper(targetUri: URI): String = {
+    s"""|<html>
+        |<head><meta http-equiv="refresh" content="0;url=${targetUri}" /></head>
+        |<body onload="window.location = '${targetUri}'"></body>
+        |</html>""".stripMargin
```

Comment:
Yeah, I can pull the JS into a separate script tag. I'm pretty sure that the the `meta` tag is actually doing the redirect in the majority of cases. One SO answer suggested including the `onload` code in case that tag wasn't picked up properly.

And yeah, it's definitely the least of all evils. I considered several different solutions here, and all of them sucked, this one just sucked the least.